lib/trino-hive-formats/src/main/java/io/trino/hive/formats/esri/EsriReader.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.esri;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.google.common.io.CountingInputStream;
+import io.trino.spi.PageBuilder;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.function.LongSupplier;
+
+import static io.trino.plugin.base.util.JsonUtils.jsonFactory;
+import static java.util.Objects.requireNonNull;
+
+public class EsriReader
+        implements Closeable
+{
+    private static final JsonFactory JSON_FACTORY = jsonFactory();
+
+    private final CountingInputStream inputStream;
+    private final PageBuilder pageBuilder;
+    private final EsriDeserializer esriDeserializer;
+    private final LongSupplier rawInputPositionSupplier;
```

Comment:
Is this `LongSupplier` necessary anymore? Seems like the only usage is in `EsriReader#getBytesRead()` and that could just call `inputStream#getCount()` directly